include/sol/stack.hpp

```diff
@@ -206,7 +206,7 @@ namespace sol {
 			void raw_table_set(lua_State* L, T&& arg, int tableindex = -2) {
 				int push_count = push(L, std::forward<T>(arg));
 				sol_c_assert(push_count == 1);
-				std::size_t unique_index = static_cast<std::size_t>(luaL_len(L, tableindex) + 1u);
+				int unique_index = luaL_len(L, tableindex) + 1u;

```

Comment:
This may not work because `luaL_len` may return a `long long` on 64-bit platforms, which makes the degradation to `int` a warning too.
Perhaps instead use...

```suggestion
				std::size_t unique_index = static_cast<std::size_t>(luaL_len(L, tableindex) + 1u);
				lua_rawseti(L, tableindex, static_cast<int>(unique_index));
```

instead?

Reply:
Bah thats what I had before and I second-guessed myself 😅 , change incoming

Reply:
Well don't take me literally on this: the `lua_rawseti` should look more like:

```suggestion
				std::size_t unique_index = static_cast<std::size_t>(luaL_len(L, tableindex) + 1u);
				lua_rawseti(L, tableindex, static_cast<lua_Number>(unique_index));
```

Although, that begs the question why we don't just static_cast to `lua_Number` to begin with and not go the loosey-goosey way around...

Reply:
Actually, ignore me: I forgot Lua 5.1 and 5.2 use `int n` for the `lua_rawseti` parameter. What you've posted is good enough: thank you!